client/components/Content/Content.jsx

```diff
@@ -0,0 +1,141 @@
+import { h, Component } from "preact";
+import cx from "classnames";
+
+import PrettyPrint from "../PrettyPrint";
+
+import styles from "./content.css";
+
+class Content extends Component {
+  constructor(...args) {
+    super(...args);
+
+    this.state = {
+      activeItem: 0
+    };
+
+    this.handleItemChange = this.handleItemChange.bind(this);
+    this.shallowDiffers = this.shallowDiffers.bind(this);
+  }
+
+  componentDidUpdate(nextProps) {
+    if (this.shallowDiffers(this.props, nextProps)) {
+      this.setState({
+        activeItem: 0
+      });
+    }
+  }
+
+  shallowDiffers(a, b) {
+    for (let i in a) if (!(i in b)) return true;
+    for (let i in b) if (a[i] !== b[i]) return true;
+    return false;
+  }
+
+  handleItemChange(index, { el, targetEl }) {
+    if (index === 0) return;
+    this.setState(
+      {
+        activeItem: index
+      },
+      () => {
+        setTimeout(() => {
+          const { top } = el.getBoundingClientRect();
+          const { height } = targetEl.getBoundingClientRect();
+          window.scrollTo(0, top - height - 40);
+        }, 0);
+      }
+    );
+  }
+
+  render(
+    {
+      data,
+      codeLength = 40, // Lines after which entire code should not be displayed
+      codeThreshold = 3, // Lines to show before and after the slice
+      baseLine = 1 // Starting line number
+    },
+    { activeItem }
+  ) {
+    if (!data || !data.length === 0) {
```

Comment:
Use default assignment instead of `!data` and there should be no `!` before `data.length === 0`

```js
render({ data = {} });
```